### DISCUSSION TITLE
How to modify content of "Set up" and "Tear down" within the Execution (Suites)

### DISCUSSION THREAD

**Discussion by dariuszkanicki** [allure-framework/allure2#2665] [Questions & Support] [answered]
I'd like to change the content of those two sections.
The "Test body" is generated based on the cucumber annotations and is not "technical".
Accordingly I want to replace the content of the "Set up" and "Tear down" section with more non-technical information instead of having the method name (incl. signature) displayed.
![Screenshot 2024-08-05 100128](https://github.com/user-attachments/assets/88bf7e66-8f3f-4e35-9691-0ee1c0b1bf6d)

**Comment by dariuszkanicki** [comment]
The final result should look like the picture below.
This means, I should be able to create those "set up" and "tear down" steps from within the Hooks class with the same "success" (or fail) icons like in the test body. 

![result1](https://github.com/user-attachments/assets/42ab42ab-f893-43c1-9d2c-7f8c32cc4925)

**Reply by sujith0980** [reply]
Hello @dariuszkanicki I have created project with Playwright cucumber using JavaScript with allure integration. I have added hooks but not adding those hooks to the allure report. I have looked at your allure report and it is very promising . Could you please provide a sample repo that includes every file for the setup ? I would really appreciate it if you can help me.

**Comment by dariuszkanicki** [comment]
At the moment, that's the temporary solution (not really user friendly):

![temp](https://github.com/user-attachments/assets/0bf64a5e-89f3-4978-8b50-36bce21491cd)

**Comment by baev** [comment, marked as answer]
By default, cucumber doesn't provide a way to change hook names. It's already introduced in the latest version of the cucumber protocol, but it is not yet available for Java. See https://github.com/cucumber/messages/issues/73 

As the solution, you can update the hook names using Allure Runtime API:

```java
@Before("@hooks")
public void before() {
    Allure.getLifecycle().updateFixture(hook -> hook.setName("Human readable before name"));
    // do something
}
```

<img width="326" alt="Screenshot 2024-08-07 at 12 38 07" src="https://github.com/user-attachments/assets/4df02a83-8283-460b-b443-2478511eb85d">

PS: Make sure you're using the latest version of `allure-cucumber7-jvm` (`2.29.0` as of today) since I remember fixes were recently made for this particular use case.

**Reply by dariuszkanicki** [reply]
That's a step forwards, thanks a lot for your hint. 
Now I got better naming (see below), however, I still would like to be able to create "step like" lines within the setup/teardown area.
At the moment I'm using text attachments, which isn't really ideal, as I would like to have proper steps with "pass/fail" icon as well.
Is that somehow doable?
![temp](https://github.com/user-attachments/assets/4c7d0a92-592a-430f-91d5-a706606bc97f)

**Reply by baev** [reply]
You can use `Allure.step` method for that:

```java
Allure.step("opened browser " + page.getUrl());
```

However, it's better to create a separate discussion per request.

**Reply by dariuszkanicki** [reply]
Thanks, that has solved everything I was looking for.

**Comment by sujith0980** [comment]
Hello @dariuszkanicki I have created project with Playwright cucumber using javascript with allure integration. I have added hooks but not adding those hooks to the allure report. I have looked at your allure report and it is very promising . Could you please provide a sample repo that includes every file for the setup ? I would really appreciate it if you can help me.
